pinc/misc.inc

```diff
@@ -153,37 +174,87 @@ function get_enumerated_param($arr, $key, $default, $choices, $allownull = false
     }
 }
 
+/**
+ * Get and validate an integer from an array
+ *
+ * If `$arr[$key]` is defined and is the string-rep of an integer
+ * satisfying certain simple constraints, return that integer.
+ * If it's not defined, and `$default` is non-null, return `$default`.
+ * If it's not defined, and `$default` is null and `$allownull` is true, return null.
+ *
+ * Example usages:
+ * - If the script is not given a 'per_page' parameter via the $_GET array,
+ *   use a default value of 20. If it is supplied, ensure it's an integer
+ *   and that it's at least 1, with no upper limit.
+ *   ```
+ *   $per_page = get_integer_param($_GET, 'per_page', 20, 1, NULL );
+ *   ```
+ * - As before, except if no limit is provided, return null, so the caller can
+ *   know to display all pages rather than some arbitrary maximum.
+ *   ```
+ *   $per_page = get_integer_param($_GET, 'per_page', null, 1, null, true);
+ *   ```
+ *
+ * @param array $arr
+ * @param string $key
+ * @param integer|null $default
+ * @param integer|null $min
+ * @param integer|null $max
+ * @param boolean $allownull
```

Comment:
Having our doc convention match the PHP type conventions makes great sense and I put up a PR to standardize on `int` and `bool` over `integer` and `boolean`. phpDocumentor supports both so that's an easy fix. Thanks for the suggestion!